Core/include/Acts/Utilities/Grid.hpp

```diff
@@ -465,20 +477,61 @@ class Grid final {
     return current_size;
   }
 
+  /// @brief Convenience function to convert the type of the grid
+  /// to hold another object type.
+  ///
+  /// @tparam U the new grid value type
+  ///
+  /// @return a new grid with the same axes and a different value type
+  template <typename U>
+  Grid<U, Axes...> convertType() const {
+    Grid<U, Axes...> cGrid(m_axes);
+    return cGrid;
+  }
+
+  /// @brief Convenience function to convert the type of the grid
+  /// to hold another object type.
+  ///
+  /// @tparam U the new grid value type
+  /// @tparam CT the converter type
+  ///
+  /// @return a new grid with the same axes and a different value type
+  template <typename U, typename CT>
+  Grid<U, Axes...> convertGrid() const {
+    Grid<U, Axes...> cGrid(m_axes);
+    // Loop through the values and convert them
+    for (std::size_t i = 0; i < size(); i++) {
+      cGrid.at(i) = CT::convert(at(i));
+    }
+    return cGrid;
+  }
+
+  /// @brief get the axes as a tuple
+  const std::tuple<Axes...> axesTuple() const { return m_axes; }
```

Comment:
This returns a copy, so there is no need imho to return it as a const object. Or did you intend to return a const reference?

The new grid constructor that you are adding will as well make a copy of the tuple, so you may end up with two copies instead of one